app/controllers/session_controller.rb

```diff
@@ -4,7 +4,8 @@
 class SessionController < ApplicationController
 
   skip_before_filter :redirect_to_login_if_required
-  skip_before_filter :preload_json, :check_xhr, only: ['sso', 'sso_login', 'become', 'sso_provider', 'destroy']
+  skip_before_filter :preload_json, :check_xhr, only: ['sso', 'sso_login', 'become', 'sso_provider', 'destroy', 'email_login_page', 'email_login']
+  skip_before_filter :check_xhr, only: ['email_login_page']
```

Comment:
Two points to note here:

1) `skip_before_filter :check_xhr` is getting defined twice above.
2) we should not skip `check_xhr` for `email_login`

I believe we should replace above with:

```
skip_before_filter :preload_json, only: ['sso', 'sso_login', 'become', 'sso_provider', 'destroy', 'email_login_page', 'email_login']
skip_before_filter :check_xhr, only: ['sso', 'sso_login', 'become', 'sso_provider', 'destroy', 'email_login_page']
```